content/articles/multiclass-text-classification-with-pyspark/index.md

```diff
@@ -0,0 +1,730 @@
+
+#### [PySpark.MLib](https://spark.apache.org/docs/latest/ml-guide.html)
+
+Contains high-level API built on top of RDD in building machine learning models. It consists of common learning algorithms for regression, classification, clustering, and collaborative filtering.
+
+In this tutorial, we shall use `PySpark.ML API` in building our multiclass text classification model.
+
+> NOTE: We are using `PySpark.ML API` in building our model because `PySpark.MLib` is deprecated and will be removed in the next PySpark releases.
+
+To further read about the components of PySpark and how it’s useful in processing [Big Data](https://en.wikipedia.org/wiki/Big_data) [click here](/engineering-education/introduction-to-spark/)
+
+### PySpark Installation
+
+We install PySpark by creating a virtual environment that keeps all the dependencies required for our project.
+Before we install PySpark we need to have `Pipenv` in our machine and we install it using the following command.
+
+```python
+pip install pipenv
+```
+
+We can now install PySpark using this command.
+
+```python
+pipenv install pyspark
+```
+
+Since we are using [Jupyter Notebook](https://jupyter.org/) in this tutorial we install it using the following command:
+
+```python
+pipenv install jupyterlab
+```
+
+In order to launch PySpark, use this command:
+
+```python
+pipenv run pyspark
+```
+
+This will launch PySpark to show if it is successfully installed in our machine.
+
+Let's activate virtual environment we have created.
+
+```python
+pipenv shell
+```
+
+To launch our notebook use this command.
+
+```python
+pipenv run jupyter lab
+```
+
+This will launch the notebook and from here we can start working on our model.
+
+### Creating SparkContext and SparkSession
+In this tutorial, we will be building a multiclass text classification model that can predict the subject category given a course title or text. We will use the Udemy dataset in the building of our model.
+
+Let's import our machine learning packages.
+
+```python
+import SparkContext from pyspark
+```
+
+`SparkContext` creates an entry point of our application and creates a connection between the different clusters in our machine allowing communication between them.
+
+The 'SparkContext' will also give a user interface that will show us all the jobs running. The `master` option specifies the master URL for our distributed cluster which will run locally. We also specify the number of threads to `2`. This allows our program to run `2` threads concurrently, thus, reduces failure of our program.
+
+```python
+sc = SparkContext(master="local[2]")
+```
+
+To launch the UI using the following command.
+
+You then click `SparkUI` to launch the Spark dashboard on `http://192.168.0.6:4040/`. This will show the available jobs running on our machine. Currently, we have no running jobs as shown:
+
+![PySpark UI](/engineering-education/multiclass-text-classification-with-pyspark/pyspark-ui.jpg)
+
+```python
+sc
+```
```

Comment:
Which command is used to launch the UI? It's not clear. 

The `sc` command also seems misplaced. What does it do? Maybe it will make sense when you reorder those few sentences